python/ray/tests/test_gcs_fault_tolerance.py

```diff
@@ -975,6 +976,62 @@ def test_redis_logs(external_redis):
         )
 
 
+@pytest.mark.parametrize(
+    "ray_start_regular_with_external_redis",
+    [
+        generate_system_config_map(
+            gcs_failover_worker_reconnect_timeout=20,
+            gcs_rpc_server_reconnect_timeout_s=60,
+        )
+    ],
+    indirect=True,
+)
+def test_job_finished_after_gcs_server_restart(ray_start_regular_with_external_redis):
+    gcs_address = ray._private.worker.global_worker.gcs_client.address
+    client = JobSubmissionClient(gcs_address)
+
+    # submit job
+    job_id = client.submit_job(
```

Comment:
```
set the gcs_rpc_server_reconnect_timeout_s = ?
submit a long running job with 1 head node
cluster.remove_node(head)
Wait until driver pid is gone
Restart head node, cluster.add_node()
make sure driver is dead
```

Reply:
changed to this flow, but still failing when trying to check if job is marked as FAILED after raylet killed, so might need help on that

Reply:
I think instead of head_node.kill_raylet, we can kill the entire head node by `cluster.remote_node(head_node)` and then restart the head node by `cluster.add_node()`